lib/lambda/item.ts

```diff
@@ -16,50 +13,51 @@ export const getItemData = async (event: APIGatewayEvent) => {
       body: { message: "Event body required" },
     });
   }
+
   try {
     const body = JSON.parse(event.body);
-    const stateFilter = await getStateFilter(event);
+
     const packageResult = await getPackage(body.id);
+    if (packageResult === undefined || !packageResult.found) {
+      return response({
+        statusCode: 404,
+        body: { message: "No record found for the given id" },
+      });
+    }
+
+    const stateFilter = await getStateFilter(event);
+    if (
```

Comment:
moved the stateFilter checks up so that we aren't calling getAppkChildren unnecessarily